### DISCUSSION TITLE
Documentation unclear on how to start the daemon

### DISCUSSION THREAD

**Discussion by laurenceSaes** [JasperFx/marten#2325] [Q&A]
The documentation contains a [page](https://martendb.io/events/projections/async-daemon.html) on how to start the background daemon. However, it only describes how this could be done in an asp.net environment. How could the daemon be started when you retrieve your own DocumentStore:

```
    var store = DocumentStore.For(options =>
    {
      options.Connection("host=***;database=***;password=***;username=***");
      options.AutoCreateSchemaObjects = AutoCreate.All;
      options.Events.AddEventTypes(new[] {
        typeof(ConfigurationRecordUpdate)
      });
      options.Projections.SelfAggregate<ConfigCollectionProjection>(ProjectionLifecycle.Async);
    });
```

I tried the example below. It appears that the daemon does not process any event updates in my projections. 

```
    var daemon = await store.BuildProjectionDaemonAsync();
    await daemon.StartDaemon();
```

At the moment, I only can use projections when I use a `ProjectionLifecycle.Inline` strategy.

**Comment by jeremydmiller** [comment]
@laurenceSaes Is there some reason why you're not willing to use the HostBuilder integration?

And all you need to do here is call `await daemon.StartAllShards()` above. If you'll notice the API comments on `StartDaemon()`, it only starts the "high water detection" and you have to start the projections independently (but starting a shard will automatically start the high water detection).
